format/Message.fbs

```diff
@@ -84,6 +84,10 @@ table Field {
   name: string;
   nullable: bool;
   type: Type;
+  // present only if the field is dictionary encoded
```

Comment:
One thing that isn't immediately obvious to me is the exact structure of the DictionaryBatch message, so it would be good to clarify this where there is that TODO. Is each DictionaryBatch constrained to only contain the dictionary for a single vector/array? If a DictionaryBatch can contain the dictionaries for multiple arrays (which seems complicated) then some more specification is needed.